File: rest/src/main/java/io/sparkled/rest/RestApiServer.java
package io.sparkled.rest;

/**
 * A REST API Server that listens on the given port.
 */
public interface RestApiServer {

    /**
     * Start the REST API server on the provided port. This method is idempotent.
     *
     * @param port The port to listen on.
     */
    void start(int port) throws Exception;
}

File: rest/src/main/kotlin/io/sparkled/rest/RestApiServer.kt
package io.sparkled.rest

/**
 * A REST API Server that listens on the given port.
 */
interface RestApiServer {

    /**
     * Start the REST API server on the provided port. This method is idempotent.

     * @param port The port to listen on.
     */
    fun start(port: Int)
}
